packages/block-editor/src/components/media-placeholder/index.js

```diff
@@ -387,10 +388,15 @@ export class MediaPlaceholder extends Component {
 
 	render() {
 		const {
+			disableMediaSelection,
 			dropZoneUIOnly,
 		} = this.props;
 
-		if ( dropZoneUIOnly ) {
+		if ( dropZoneUIOnly || disableMediaSelection ) {
```

Comment:
I think we have a bug here, even if the new prop disableMediaSelection is used we always call the deprecated function. It should be called if dropZoneUIOnly is used but not called if disableMediaSelection is used.